backend/app/services/splunk.py

```diff
@@ -116,24 +111,15 @@ async def filterPost(self, query, searchList=""):
 
         try:
             # If additional search parameters are provided, include those in searchindex
-            searchindex = (
-                f"search index={self.indice} {searchList}"
-                if searchList
-                else f"search index={self.indice}"
-            )
             search_query = ""
             if searchList:
-                search_query = "search index={} {} | stats count AS total_records, values(cpu) AS cpu, values(node_name) AS nodeName, values(test_type) AS benchmark, values(ocp_version) AS ocpVersion, values(ocp_build) AS releaseStream".format(
-                    self.indice, searchList
-                )
+                search_query = f"search index={self.indice} {searchList} | stats count AS total_records, values(cpu) AS cpu, values(node_name) AS nodeName, values(test_type) AS benchmark, values(ocp_version) AS ocpVersion, values(ocp_build) AS releaseStream"
             else:
-                search_query = "search index={} | stats count AS total_records, values(cpu) AS cpu, values(node_name) AS nodeName, values(test_type) AS benchmark, values(ocp_version) AS ocpVersion, values(ocp_build) AS releaseStream".format(
-                    self.indice
-                )
+                search_query = f"search index={self.indice} | stats count AS total_records, values(cpu) AS cpu, values(node_name) AS nodeName, values(test_type) AS benchmark, values(ocp_version) AS ocpVersion, values(ocp_build) AS releaseStream"
 
             try:
                 # Run the job and retrieve results
-                job = await self.service.jobs.create(
+                job = self.service.jobs.create(
```

Comment:
Above where you removed the `await`, you replaced it with a `not job.is_done()` busy loop. I think we discussed a similar change earlier, though I don't remember exactly why it was better ... but more importantly you don't seem to be doing anything here to wait for `job` before referencing the results at line 134 ... won't that fail?

Reply:
Yeah reverted the change as `await` in that line is throwing error.

Reply:
Ah, OK -- because the `splunklib` `Jobs.create()` doesn't return a Python `Coroutine` at all, so `await` won't work. But the semantics of the Splunk `Job.results()` isn't entirely clear in the documentation I saw. Is the `while not job.is_done()` loop above necessary? And, if so, don't you also need that here? `Job.results()` returns a "stream", so it's quite possible that the explicit wait loop *isn't* necessary, and we'll just block at the read until there are results, reading them one-by-one. And if that works here, would it work up above at line 66/67 as well instead of the explicit wait loop???

Reply:
If there is no wait loop, then oneshot is not streaming the results, causing the query to return only the `total_records` value.

Reply:
I think maybe that makes sense to me. Although I'm not sure whether I need to learn more about Splunk or whether I want to run and hide. 😆 

So the wait loop above is because you're using the `oneshot` to stream results, whereas here you're just iterating on `job.results` directly. Huh. I'll take your word for it, but thanks for the patient explanation. 😆